applications/unit_tests/helpers.hpp

```diff
@@ -7,9 +7,64 @@
 
 #include <doctest/doctest.h>
 #include <llri/llri.hpp>
+#include <sstream>
 
 namespace detail
 {
+    /**
+     * @brief Utility function for hashing strings  in compile time
+    */
+    template<size_t N>
+    inline constexpr uint64_t nameHash(const char(&name)[N]) noexcept
+    {
+        uint64_t hash = 0xcbf29ce484222325;
+        constexpr uint64_t prime = 0x00000100000001b3;
+
+        for (size_t i = 0; i < N; i++)
+        {
+            if (name[i] == '\0')
+                break;
+            
+            hash = hash ^ static_cast<uint8_t>(name[i]);
+            hash *= prime;
+        }
+
+        return hash;
+    }
+
+    /**
+     * @brief Utility function for hashing std::strings in runtime
+    */
+    inline uint64_t nameHash(const std::string& name)
+    {
+        uint64_t hash = 0xcbf29ce484222325;
+        constexpr uint64_t prime = 0x00000100000001b3;
+
+        for (size_t i = 0; i < name.size(); i++)
+        {
+            if (name[i] == '\0')
+                break;
+            
+            hash = hash ^ static_cast<uint8_t>(name[i]);
+            hash *= prime;
+        }
+
+        return hash;
+    }
+
+    inline std::unordered_map<uint64_t, std::string> split (const std::string &s, char delim)
```

Comment:
since it does more than just split, you might want to call it something  different now?

Reply:
true, not sure what to call it tho 🤔